fix(sdk): permit empty local execution outputs #localexecution

connor-mccarthy opened the pull request:
**Description of your changes:**
It should be permitted for a user to specify a `dsl.OutputPath` to which the output is not written. This PR fixes a key error when this is the case.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 
<!--
   PR titles examples:
    * `fix(frontend): fixes empty page. Fixes #1234`
       Use `fix` to indicate that this PR fixes a bug.
    * `feat(backend): configurable service account. Fixes #1234, fixes #1235`
       Use `feat` to indicate that this PR adds a new feature. 
    * `chore: set up changelog generation tools`
       Use `chore` to indicate that this PR makes some changes that users don't need to know.
    * `test: fix CI failure. Part of #1234`
        Use `part of` to indicate that a PR is working on an issue, but shouldn't close the issue when merged.
-->

connor-mccarthy commented:
/test all

connor-mccarthy commented:
/assign @chensun

chensun reviewed (commented):
It seems there's no fix, but only adding test? If so, title can be `test(sdk): ...`

/lgtm

connor-mccarthy commented:
Good catch, @chensun. Updated.

/approve

connor-mccarthy commented:
@chensun, NVM. Reverted title per https://github.com/kubeflow/pipelines/pull/10339/files#r1440864867. Also updated the test case so that it fails before the corresponding change, but passes after.

/remove-approve

chensun reviewed (approved):
/lgtm
/approve

Thanks!

connor-mccarthy commented:
/retest

connor-mccarthy commented:
/retest

connor-mccarthy commented:
/retest

connor-mccarthy commented:
/test kubeflow-pipelines-sdk-execution-tests
